addons/bus/static/src/js/services/websocket_service.js

```diff
@@ -0,0 +1,123 @@
+/** @odoo-module **/
+import { browser } from "@web/core/browser/browser";
+import { registry } from '@web/core/registry';
+
+import { WebsocketError } from '@bus/js/websocket_errors';
+
+const { EventBus } = owl;
+/**
+ * Communicate with a SharedWorker in order to provide a single websocket
+ * connection shared across multiple tabs.
+ *
+ *  @emits connect
+ *  @emits disconnect
+ *  @emits reconnect
+ *  @emits reconnecting
+ *  @emits notification
+ */
+export const websocketService = {
+    dependencies: ['cookie', 'notification', 'user'],
+    _removeConnectionLostNotification: null,
+
+    start(env) {
+        this.env = env;
```

Comment:
you should not use `this` here. It's not a class. Just create variables in the closure: `const bus = new EventBus();`

Reply:
note that looking at the code, it's rather nice